File: SimpleImageViewer/SimpleImageViewerJVM/src/main/java/SimpleImageViewerJVM/Picture.java
package SimpleImageViewerJVM;

import java.awt.Color;
import java.awt.image.BufferedImage;

import com.spvessel.spacevil.*;
import com.spvessel.spacevil.Flags.*;

public class Picture extends Prototype implements Comparable<Picture> {

    public ImageItem image = new ImageItem();
    public Label name = new Label();
    public String path = "";

    public Picture(BufferedImage image, String name, String path) {
        setSizePolicy(SizePolicy.EXPAND, SizePolicy.EXPAND);
        setSpacing(0, 5);
        setMargin(2, 2, 2, 2);
        setAlignment(ItemAlignment.LEFT, ItemAlignment.TOP);
        setBackground(new Color(80, 80, 80));
        setShadow(10, 0, 0, new Color(0, 0, 0, 200));

        this.image.setImage(image);
        this.name.setText(name);
        this.path = path;
    }

    @Override
    public int compareTo(Picture item) {
        return this.path.compareTo(item.path);
    }

    @Override
    public void initElements() {
        name.setHeightPolicy(SizePolicy.FIXED);
        name.setHeight(30);
        name.setTextAlignment(ItemAlignment.HCENTER, ItemAlignment.VCENTER);
        name.setMargin(5, 0, 5, 0);
        name.setAlignment(ItemAlignment.LEFT, ItemAlignment.BOTTOM);

        image.isHover = false;
        image.keepAspectRatio(true);
        image.setBackground(32, 32, 32);
        image.setMargin(0, 0, 0, 35);

        addItems(image, name);
    }
}

File: Apps/Demos/SimpleImageViewer/SimpleImageViewerNet/Picture.cs
using System;
using System.Drawing;
using SpaceVIL.Common;
using SpaceVIL.Decorations;
using SpaceVIL.Core;
using SpaceVIL;
using System.IO;

namespace SimpleImageViewer
{
    public class Picture : Prototype, IComparable<Picture>
    {
        public ImageItem Image = new ImageItem();
        public Label Name = new Label();
        public String Path = String.Empty;

        public Picture(Bitmap image, String name, String path)
        {
            SetSizePolicy(SizePolicy.Expand, SizePolicy.Expand);
            SetSpacing(0, 5);
            SetMargin(2, 2, 2, 2);
            SetAlignment(ItemAlignment.Left, ItemAlignment.Top);
            SetBackground(Color.FromArgb(80, 80, 80));
            SetShadow(10, 0, 0, Color.FromArgb(200, 0, 0, 0));

            Image.SetImage(image);
            Name.SetText(name);
            Path = path;
        }

        public int CompareTo(Picture item)
        {
            return this.Path.CompareTo(item.Path);
        }

        public override void InitElements()
        {
            Name.SetHeightPolicy(SizePolicy.Fixed);
            Name.SetHeight(30);
            Name.SetTextAlignment(ItemAlignment.HCenter, ItemAlignment.VCenter);
            Name.SetMargin(5, 0, 5, 0);
            Name.SetAlignment(ItemAlignment.Left, ItemAlignment.Bottom);

            Image.IsHover = false;
            Image.KeepAspectRatio(true);
            Image.SetBackground(32, 32, 32);
            Image.SetMargin(0, 0, 0, 35);

            AddItems(Image, Name);
        }
    }
}
